cmd/lifecycle/detector.go

```diff
@@ -53,34 +63,75 @@ func (d *detectCmd) Privileges() error {
 }
 
 func (d *detectCmd) Exec() error {
-	group, plan, err := d.detect()
+	dr, err := d.detect()
 	if err != nil {
 		return err
 	}
-	return d.writeData(group, plan)
+	return d.writeData(dr)
 }
 
-func (da detectArgs) detect() (lifecycle.BuildpackGroup, lifecycle.BuildPlan, error) {
+func (da detectArgs) mergeOrderWithStackBuildpacks(order lifecycle.BuildpackOrder) (lifecycle.BuildpackOrder, error) {
+	if _, err := os.Stat(da.stackOrderPath); err != nil {
+		if os.IsNotExist(err) {
+			return order, nil
+		}
+	}
+
+	stackBuildPacksOrder, err := lifecycle.ReadOrder(da.stackOrderPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(stackBuildPacksOrder) == 0 {
+		return order, nil
+	}
+
+	if len(stackBuildPacksOrder) > 1 {
```

Comment:
Ah, interesting - what is the motivation for this requirement? I don't see it in the RFC, but maybe I missed it.

Reply:
Yeah, I think it was just added or is going to be added. 

The reason we did this is because we would have to explode the combined list of stackpacks+buildpacks. Right now if you have 5 buildpack groups, we end up still testing just 5 groups by prepending the stackpacks. If you had 3 groups of stackpacks, I think we would need to create a group for each stackpack group combined with each buildpack group. Maybe that isn't bad and there is a use case, but we couldn't think of one.

Reply:
I am all for not exploding the number of scenarios we need to consider, if it's not necessary!